File: src/api/log.rs
use mlua::prelude::*;
use mooncake::mooncake;
use tracing::{debug, error, info, trace, warn};

use crate::{api::get_plugin_id, package};

package! {
    trace, debug, info, warn, error
}

#[mooncake(lua)]
fn trace(msg: String) -> LuaResult<()> {
    trace!("[{}]: {msg}", get_plugin_id(lua)?);
    Ok(())
}
#[mooncake(lua)]
fn debug(msg: String) -> LuaResult<()> {
    debug!("[{}]: {msg}", get_plugin_id(lua)?);
    Ok(())
}
#[mooncake(lua)]
fn info(msg: String) -> LuaResult<()> {
    info!("[{}]: {msg}", get_plugin_id(lua)?);
    Ok(())
}
#[mooncake(lua)]
fn warn(msg: String) -> LuaResult<()> {
    warn!("[{}]: {msg}", get_plugin_id(lua)?);
    Ok(())
}
#[mooncake(lua)]
fn error(msg: String) -> LuaResult<()> {
    error!("[{}]: {}", get_plugin_id(lua)?, msg);
    Ok(())
}

File: libs/rsa-core/src/api/lua/lib/log.lua
-- =============== Logging ===============
-- Docs are forwarded from rust log library

--- Logs on the "trace" level.
---
--- Designates very low priority, often extremely verbose, information.
--- @param msg string
function trace(msg) end

--- Logs on the "debug" level.
---
--- Designates lower priority information.
--- @param msg string
function debug(msg) end

--- Logs on the "info" level.
---
--- Designates useful information.
--- @param msg string
function info(msg) end

--- Logs on the "warn" level.
---
--- Designates hazardous situations.
--- @param msg string
function warn(msg) end

--- Logs on the "error" level.
---
--- Designates very serious errors.
--- @param msg string
function error(msg) end